### PR TITLE
Handle Mac OS X zipped files by ignoring the OS X specific sidecar di…

### PR DESCRIPTION
**Pull request by travislbrundage**
…rectory

### CODE REVIEW

File: osgeo_importer/forms.py [hunk 1]
```diff
@@ -41,6 +41,9 @@ def clean(self):
             if is_zipfile(f):
                 with ZipFile(f) as zip:
                     for zipname in zip.namelist():
+                        # Skip mac directory sidecar
+                        if zipname == '__MACOSX/':
+                            continue
```
Comment (travislbrundage) [inline osgeo_importer/forms.py]: Not sure if there's a smarter way to do this, but in the end this seemed to be the 'simplest' way - just ignore this directory which is always present in files that were compressed in OS X.

I don't like that it's kind of ad hoc, but I haven't come up with a better solution. Open to suggestions.

**Comment by Coop56** [comment]
Could you send me a dataset that is effected by this?  I would think our valid extensions list would check the files in this directory and ignore them already if they don't match the file types we look for.

**Comment by JivanAmara** [comment]
This issue has definitely been addressed (though it's possible a regression has unaddressed it).  I'm pretty sure that @Coop56 is right and the valid/ignore framework should alread handle it.

**Comment by travislbrundage** [comment]
The problem is it chokes on a directory - there is no file extension, so it doesn't end up checking.

This should be possible to replicate by compressing any data on a Mac and trying to import it. It'll end up failing on this line: https://github.com/GeoNode/django-osgeo-importer/blob/master/osgeo_importer/forms.py#L44

Because it is trying to read an extension that isn't there, instead capturing the name of a subdirectory, '__MACOSX', which is something OS X creates in any compressed archive for some reason.

The error:
image
image

**Comment by Coop56** [comment]
It's possible (likely) that we will run into other issues with directories within zipped shapefiles so maybe instead of excluding that one directly we can have the form validation walk through directories and add / exclude the necessary files.

**Comment by travislbrundage** [comment]
Agreed - I will make this more agnostic.

Should we just recurse into the subdirectories and validate the files within it as well?

**Comment by bitner** [comment]
I believe that it does recurse already to get the files. What might make sense is to have the clean step "flatten" all the files to end up with a directory with just the files from any nested directories without any directories.

**Comment by jj0hns0n** [comment]
Flattening the directories could be fine as long as base names that are the same have the dir name appended. I can imagine having several sets of roads.shp files each in a different directory.

**Comment by travislbrundage** [comment]
I'll go with that and update this PR accordingly. Thanks all for the feedback!

**Pull request closed without merging**